Kentico.Kontent.Delivery/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -186,6 +188,8 @@ private static IServiceCollection RegisterDependencies(this IServiceCollection s
             services.TryAddSingleton<IRetryPolicyProvider, DefaultRetryPolicyProvider>();
             services.TryAddSingleton<IDeliveryClient, DeliveryClient>();
             services.TryAddSingleton<IDeliveryClientFactory, DeliveryClientFactory>();
+            services.TryAddSingleton<IDeliveryCacheManager, DeliveryCacheManager>();
+            services.TryAddSingleton<IMemoryCache, MemoryCache>();
```

Comment:
If we use decorators, we can avoid registering default caching implementations every time. We can do that only when needed - in `WithDeliveryCacheManager()`.